lib/faker/default/company.rb

```diff
@@ -464,6 +464,59 @@ def sic_code
         fetch('company.sic_code')
       end
 
+      ##
+      # Get a random Indian Goods and Services Tax (GST) number.
+      # For more on Indian tax number here:
+      # https://simple.wikipedia.org/wiki/GSTIN
+      # @params state code [String] Any state code.
+      #
+      # @return [String]
+      # @example
+      #   Faker::Company.indian_gst_number #=> "15VQPNZ2126J2ZU"
+      #   Faker::Company.indian_gst_number(state_code: "22") #=> "22ZVWEY6632K0ZN"
+      #
+      # @faker.version 3.2.1
+      def indian_gst_number(state_code: nil)
+        # Check if state code is valid
+        state_code_ranges = [('02'..'38'), ['98']]
+        raise ArgumentError, 'state code must be in a range of 02 to 38 or 98' if state_code && !(state_code_ranges[0].include?(state_code) || state_code == '98')
```

Comment:
```suggestion
        if state_code && !(state_code_ranges[0].include?(state_code) || state_code == '98')
          raise ArgumentError, 'state code must be in a range of 02 to 38 or 98'
        end
```

What do you think of rewriting it like this? It helps understanding more what the intent is.